### PR TITLE
Edit `influxd upgrade` docs

### LINKED ISSUE #1680: Upgrade guide 1.8 -> 2.0 rc1
InfluxDB 2 RC 1 introduces the ability to upgrade from InfluxDB 1.x to InfluxDB 2 performed by the `influxd upgrade` command.

```
> influxd upgrade --help

    Upgrades a 1.x version of InfluxDB by performing the following actions:
      1. Reads 1.x config file and creates 2.x config file with matching options. Unsupported 1.x options are reported.
      2. Copies and upgrades 1.x database files.
      3. Creates a script for creating 1.x users and their permissions. This script needs to be revised and run manually after starting 2.x.

    If the config file is not available, 1.x db folder (--v1-dir options) is taken as an input.
    Target 2.x database dir is specified by the --engine-path option. If changed, the bolt path should be changed as well.

Usage:
  influxd upgrade [flags]
  influxd upgrade [command]

Available Commands:
  v1-dump-meta Dump InfluxDB 1.x meta.db
  v2-dump-meta Dump InfluxDB 2.x influxd.bolt

Flags:
  -m, --bolt-path string             path for boltdb database (default "/<user-home>/.influxdbv2/influxd.bolt")
  -b, --bucket string                primary bucket name
      --config-file string           optional: Custom InfluxDB 1.x config file path, else the default config file
  -e, --engine-path string           path for persistent engine files (default "/<user-home>/.influxdbv2/engine")
  -f, --force                        skip confirmation prompt
  -h, --help                         help for upgrade
      --influx-command-path string   path to influx command (default "/usr/bin/influx")
      --log-path string              optional: custom log file path (default "/<user-home>/upgrade.log")
  -o, --org string                   primary organization name
  -p, --password string              password for username
  -r, --retention string             optional: duration bucket will retain data. 0 is infinite. Default is 0.
      --security-script string       optional: generated security upgrade script path (default "/<user-home>/influxd-upgrade-security.sh")
  -t, --token string                 optional: token for username, else auto-generated
  -u, --username string              primary username
      --v1-dir string                path to source 1.x db directory containing meta, data and wal sub-folders (default "/<user-home>/.influxdb")
  -v, --verbose                      verbose output (default true)
```
**Prerequisities**
 - Stopped InfluxDB 1.x.
 - Installed, but not started and not initialized InfluxDB 2 rc1 and above. If InfluxDB 2 has already been initialized (created admin user), either the data files (`engine` folder) and the bolt file (`influxdb.bolt`) must be deleted or upgrade must be run against different folder and bolt file, which they must be user later when starting server. E.g.:
`influxd upgrade --engine-path /mnt/data/influxdb2/engine --bolt-path /mnt/data/influxdb2/influxdb.bolt`
After upgrade:
`influxd  --engine-path /mnt/data/influxdb2/engine --bolt-path /mnt/data/influxdb2/influxdb.bolt`
  
**Running upgrade**
Important paths to validate: 
 - InfluxDB 1 config file (--config-file): Config file of the InfluxDB 1 server. Path to data dirs (Data, Wal, and Meta) is read from it and all supported options are written to new config for InfluxDB 2. Default `/etc/influxdb/influxdb.conf`.
- InfluxDB 2 engine path (-e, --engine-path): Path for persistent InfluxDB 2 engine files. Default `/<user-home>/.influxdbv2/engine`. It must be empty or not exist before starting upgrade.
- InfluxDB 2 bolt path (-b, --bolt-path):  Path for InfluxDB 2  boltdb database.  Default `/<user-home>/.influxdbv2/influxd.bolt`. It must not exist before starting upgrade.

run: `influxd upgrade`, or with additional options.

When the upgrade is started and no flags are specified, the default paths are considered as inputs and a user is interactively asked to enter org, user name, user password, bucket, and retention parameters.

If InfuxDB 1 contained users with defined privileges, influxd upgrade will create a script that will allow creating authorization tokens with privileges to the same data. This script should be revised and that run after starting the server with upgraded data.

**After the upgrade**
The new config file can be applied in two ways:
 - Use env var: Set config file path to INFLUXD_CONFIG_PATH env var.
-  Place config file in the same folder ad `influxd`. The config file must have name 'config.yaml`.

Start the server.

Run the script for creating privileges.

### PR DESCRIPTION
**Pull request by pierwill**
Work toward closing #1680. Will follow up with a PR making any needed updates to [this guide](https://docs.influxdata.com/influxdb/v2.0/reference/upgrading/influxd-upgrade-guide/).

- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

**Comment by pierwill** [comment]
(Consider editing https://github.com/influxdata/influxdb/blob/master/cmd/influxd/upgrade/upgrade.go#L118-L124 as well.)

File: content/influxdb/v2.0/reference/cli/influxd/upgrade.md [hunk 1]
```diff
@@ -9,26 +9,50 @@ products: [oss]
...
 
 {{% warn %}}
 Be sure to back up all data before upgrading with `influx upgrade`.
 {{% /warn %}}
 
+This command performs the following actions:
+
+1. Reads 1.x configuration file and creates a 2.x config file with matching options.
+   Unsupported 1.x options are reported.
+2. Copies and upgrades 1.x database files.
```
Comment (kelseiv) [inline content/influxdb/v2.0/reference/cli/influxd/upgrade.md:24]: Think we can streamline by replacing Step 2 with lines 12-13?

File: content/influxdb/v2.0/reference/cli/influxd/upgrade.md [hunk 2]
```diff
@@ -9,26 +9,50 @@ products: [oss]
...
 [retention policies](/influxdb/v1.8/concepts/glossary/#retention-policy-rp) (used in 1.x)
 over to [buckets](/influxdb/v2.0/reference/glossary/#bucket) (used in 2.0).
 
 {{% warn %}}
 Be sure to back up all data before upgrading with `influx upgrade`.
 {{% /warn %}}
 
+This command performs the following actions:
+
+1. Reads 1.x configuration file and creates a 2.x config file with matching options.
```
Comment (kelseiv) [inline content/influxdb/v2.0/reference/cli/influxd/upgrade.md:22]: Creates the InfluxDB 2.x configuration file using 1.x configuration file options.

File: content/influxdb/v2.0/reference/cli/influxd/upgrade.md [hunk 3]
```diff
@@ -9,26 +9,50 @@ products: [oss]
...
+   Unsupported 1.x options are reported.
```
Comment (kelseiv) [inline content/influxdb/v2.0/reference/cli/influxd/upgrade.md:23]: Curious, where these options are reported?

File: content/influxdb/v2.0/reference/cli/influxd/upgrade.md [hunk 4]
```diff
@@ -9,26 +9,50 @@ products: [oss]
...
+2. Copies and upgrades 1.x database files.
+3. Creates a script for creating tokens that correspond to 1.x users.
```
Comment (kelseiv) [inline content/influxdb/v2.0/reference/cli/influxd/upgrade.md:25]: ...that creates tokens for 1.x users.

File: content/influxdb/v2.0/reference/cli/influxd/upgrade.md [hunk 5]
```diff
@@ -9,26 +9,50 @@ products: [oss]
...
+   This script needs to be revised and run manually after starting 2.x.
```
Comment (kelseiv) [inline content/influxdb/v2.0/reference/cli/influxd/upgrade.md:26]: How will folks need to revise their script?

4. Revise the script by xyz. 
5. After starting 2.x, manually run revised script.

File: content/influxdb/v2.0/reference/cli/influxd/upgrade.md [hunk 6]
```diff
@@ -9,26 +9,50 @@ products: [oss]
...
+
+If the configuration file is not available, the 1.x database folder can be passed via th `--v1-dir` flag.
```
Comment (kelseiv) [inline content/influxdb/v2.0/reference/cli/influxd/upgrade.md:28]: Think we can move this up to Step 1. 
Perhaps:
If....not available, pass the 1.x...using `insert full command`

File: content/influxdb/v2.0/reference/cli/influxd/upgrade.md [hunk 7]
```diff
@@ -9,26 +9,50 @@ products: [oss]
...
+The target 2.x database directory is specified by the `--engine-path` option.
```
Comment (kelseiv) [inline content/influxdb/v2.0/reference/cli/influxd/upgrade.md:29]: Is this used in conjunction with passing the 1.x config file? If yes, does it make sense to combine both options?
Specify the target 2.x ...

File: content/influxdb/v2.0/reference/cli/influxd/upgrade.md [hunk 8]
```diff
@@ -9,26 +9,50 @@ products: [oss]
...
+If changed, the bolt path should be changed as well.
```
Comment (kelseiv) [inline content/influxdb/v2.0/reference/cli/influxd/upgrade.md:30]: How do folks update the bolt path/is there a command?

File: content/influxdb/v2.0/reference/cli/influxd/upgrade.md [hunk 9]
```diff
@@ -9,26 +9,50 @@ products: [oss]
...
-|      | `--config`       | (Optional) Custom InfluxDB path to 1.x config file to upgrade (default: `~/.influxdbv2/conf`) | string     |
-|      | `--v1-meta-dir`  | Path to 1.x meta.db directory (default: `~/.influxdb/meta`)                                   | string     |
-|      | `--v2-bolt-path` | Path to 2.0 BoltDB database (default: `~/.influxdbv2/influxd.bolt`)                           | string     |
+| Flag |                         | Description                                                                                        | Input type |
+|:-----|:------------------------|:---------------------------------------------------------------------------------------------------|:----------:|
+| `-m` | `--bolt-path`           | Path for boltdb database (default "~/.influxdbv2/influxd.bolt")                                    | string     |
+| `-b` | `--bucket`              | Primary bucket name                                                                                | string     |
+|      | `--config-file`         | (Optional) Custom InfluxDB 1.x config file path, else the default config file                      | string     |
```
Comment (kelseiv) [inline content/influxdb/v2.0/reference/cli/influxd/upgrade.md:45]: is the default still `~/.influxdbv2/conf`?

File: content/influxdb/v2.0/reference/cli/influxd/upgrade.md [hunk 10]
```diff
@@ -9,26 +9,50 @@ products: [oss]
...
-|:-----|:-----------------|:----------------------------------------------------------------------------------------------|:----------:|
-| `-h` | `--help`         | Help for the `upgrade` command                                                                |            |
-|      | `--config`       | (Optional) Custom InfluxDB path to 1.x config file to upgrade (default: `~/.influxdbv2/conf`) | string     |
-|      | `--v1-meta-dir`  | Path to 1.x meta.db directory (default: `~/.influxdb/meta`)                                   | string     |
-|      | `--v2-bolt-path` | Path to 2.0 BoltDB database (default: `~/.influxdbv2/influxd.bolt`)                           | string     |
+| Flag |                         | Description                                                                                        | Input type |
+|:-----|:------------------------|:---------------------------------------------------------------------------------------------------|:----------:|
+| `-m` | `--bolt-path`           | Path for boltdb database (default "~/.influxdbv2/influxd.bolt")                                    | string     |
```
Comment (kelseiv) [inline content/influxdb/v2.0/reference/cli/influxd/upgrade.md:43]: think we can replace double quotes with single quotes, lines 43 down

File: content/influxdb/v2.0/reference/cli/influxd/upgrade.md [hunk 11]
```diff
@@ -9,26 +9,50 @@ products: [oss]
...
+| `-e` | `--engine-path`         | Path for persistent engine files (default "~/.influxdbv2/engine")                                  | string     |
+| `-f` | `--force`               | Skip the confirmation prompt                                                                       |            |
+| `-h` | `--help`                | Help for `influxd upgrade`                                                                         |            |
+|      | `--influx-command-path` | Path to influx command (default "~/go/bin/influx")                                                 | string     |
+|      | `--log-path`            | (Optional) custom log file path (default "~/upgrade.log")                                          | string     |
+| `-o` | `--org`                 | Primary organization name                                                                          | string     |
+| `-p` | `--password`            | Password for username                                                                              | string     |
+| `-r` | `--retention`           | (Optional) duration bucket will retain data. 0 is infinite. The default is 0.                      | string     |
```
Comment (kelseiv) [inline content/influxdb/v2.0/reference/cli/influxd/upgrade.md:53]: (Optional) Duration bucket retains... (default `0`; retains data infinitely)

File: content/influxdb/v2.0/reference/cli/influxd/upgrade.md [hunk 12]
```diff
@@ -9,26 +9,50 @@ products: [oss]
...
+|      | `--security-script`     | (Optional) generated security upgrade script path (default "~/influxd-upgrade-security.sh")        | string     |
```
Comment (kelseiv) [inline content/influxdb/v2.0/reference/cli/influxd/upgrade.md:54]: Generated

File: content/influxdb/v2.0/reference/cli/influxd/upgrade.md [hunk 13]
```diff
@@ -9,26 +9,50 @@ products: [oss]
...
+| `-t` | `--token`               | (Optional) token for username, else auto-generated                                                 | string     |
```
Comment (kelseiv) [inline content/influxdb/v2.0/reference/cli/influxd/upgrade.md:55]: Token...username. If not specified, a token is auto-generated.

File: content/influxdb/v2.0/reference/cli/influxd/upgrade.md [hunk 14]
```diff
@@ -9,26 +9,50 @@ products: [oss]
...
+| `-u` | `--username`            | Primary username                                                                                   | string     |
```
Comment (kelseiv) [inline content/influxdb/v2.0/reference/cli/influxd/upgrade.md:56]: "admin" user responsible for creating other users/tokens?

File: content/influxdb/v2.0/reference/cli/influxd/upgrade.md [hunk 15]
```diff
@@ -9,26 +9,50 @@ products: [oss]
...
+|      | `--v1-dir`              | Path to source 1.x db directory containing meta, data, and wal sub-folders (default "~/.influxdb") | string     |
```
Comment (kelseiv) [inline content/influxdb/v2.0/reference/cli/influxd/upgrade.md:57]: Path to the 1.x `db` directory....`meta`, `data`, and `wal` folders...

File: content/influxdb/v2.0/reference/cli/influxd/upgrade.md (hunk 5 above)
Comment (stuartcarnie) [reply content/influxdb/v2.0/reference/cli/influxd/upgrade.md:26]: The security script is going away when https://github.com/influxdata/influxdb/pull/19854 is completed and merged. As a result of this PR, users will automatically be created as tokens in 2.x without a script, simplifying the upgrade process significantly.

File: content/influxdb/v2.0/reference/cli/influxd/upgrade.md (hunk 8 above)
Comment (stuartcarnie) [reply content/influxdb/v2.0/reference/cli/influxd/upgrade.md:30]: The user can specify a switch with the `influxd upgrade` command:

```
  -m, --bolt-path string             path for boltdb database (default "/Users/stuartcarnie/.influxdbv2/influxd.bolt")
```

File: content/influxdb/v2.0/reference/cli/influxd/upgrade.md [hunk 16]
```diff
@@ -9,26 +9,50 @@ products: [oss]
...
+| `-h` | `--help`                | Help for `influxd upgrade`                                                                                 |            |
+|      | `--influx-command-path` | Path to influx command (default `~/go/bin/influx`)                                                         | string     |
+|      | `--log-path`            | (Optional) Custom log file path (default `~/upgrade.log`)                                                  | string     |
+| `-o` | `--org`                 | Primary organization name                                                                                  | string     |
+| `-p` | `--password`            | Password for username                                                                                      | string     |
+| `-r` | `--retention`           | (Optional) Duration bucket will retain data (default `0`; retains data infinitely)                         | string     |
+|      | `--security-script`     | (Optional) Generated security upgrade script path (default `~/influxd-upgrade-security.sh`)                | string     |
```
Comment (stuartcarnie) [inline content/influxdb/v2.0/reference/cli/influxd/upgrade.md:54]: This switch is going away when https://github.com/influxdata/influxdb/pull/19854 is completed and merged

File: content/influxdb/v2.0/reference/cli/influxd/upgrade.md (hunk 7 above)
Comment (stuartcarnie) [reply content/influxdb/v2.0/reference/cli/influxd/upgrade.md:29]: This is used if the user wishes to override the destination data path for the InfluxDB 2.x instance. Ordinarily, the default value of this argument would be fine.

File: content/influxdb/v2.0/reference/cli/influxd/upgrade.md (hunk 6 above)
Comment (stuartcarnie) [reply content/influxdb/v2.0/reference/cli/influxd/upgrade.md:28]: I would expect the use of this flag is an edge case, if the user did not have a configuration file, but yes, moving to step 1. makes sense.

File: content/influxdb/v2.0/reference/cli/influxd/upgrade.md (hunk 5 above)
Comment (pierwill) [reply content/influxdb/v2.0/reference/cli/influxd/upgrade.md:26]: Not sure, but @stuartcarnie notes that this is going away soon with https://github.com/influxdata/influxdb/pull/19854.

**Pull request merged**